partition_snapshot_reader.hh

```diff
@@ -287,9 +288,17 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
                         && !rt_less(pos, peek_range_tombstone().position())
                         && (_rt_stream.empty() || !rt_less(_rt_stream.peek_next().position(), peek_range_tombstone().position()))) {
                     range_tombstone rt = pop_range_tombstone();
-                    if (rt.trim(*_snapshot_schema,
-                                position_in_partition_view::for_range_start(ck_range),
-                                position_in_partition_view::for_range_end(ck_range))) {
+
+                    // To avoid copying ck_range even if we are not reversing, let's make this hack.
+                    std::optional<query::clustering_range> ck_range_opt;
+                    if (Reversing)
+                        ck_range_opt = query::reverse_clustering_range(ck_range);
+
+                    const query::clustering_range& ck_range_ref = ck_range_opt ? *ck_range_opt : ck_range;
```

Comment:
We should pass the native order to this reader, that's what we'll eventually end up in all readers, right?

Instead of calling clustering_key_filter_ranges::get_ranges() in the memtable layers, we should call get_native_ranges().

this way we avoid double reversal.

Reply:
> We should pass the native order to this reader, that's what we'll eventually end up in all readers, right?
Currently I expect that we don't do that. When the conversation about this design resolves, I will fix it accordingly.

> this way we avoid double reversal.
Either we reverse here or in lsa_partition_reader::do_refresh_state, but indeed, if I do the reversal in do_refresh_state, eventually double reversal will be eliminated.